### PR TITLE
zif_abapgit_html_viewer: refactor CNHT_POST_DATA_TAB & CNHT_QUERY_TABLE 

### LINKED ISSUE #4381: zif_abapgit_html_viewer: exceptions + CNHT_POST_DATA_TAB & CNHT_QUERY_TABLE
Hi,
I'd like to further decouple https://github.com/abapGit/abapGit/blob/master/src/ui/zif_abapgit_html_viewer.intf.abap

A: The exceptions given by the methods should be ZCX_ABAPGIT_EXCEPTION

B: remove uses of CNHT_POST_DATA_TAB & CNHT_QUERY_TABLE, these can be defined in the interface, and then usages removed https://github.com/abapGit/abapGit/search?q=CNHT_POST_DATA_TAB, they should only be used in ZCL_ABAPGIT_HTML_VIEWER_GUI

### PR DESCRIPTION
**Pull request by larshp**
zif_abapgit_html_viewer: refactor CNHT_POST_DATA_TAB & CNHT_QUERY_TABLE 

Note that this avoids using type definitions from type pool CNHT across the abapGit codebase, which will help to decouple from SAP standard plus enhancing static type checking by abaplint

closes #4381

### CODE REVIEW

File: src/ui/core/zif_abapgit_gui_event.intf.abap [hunk 1]
```diff
@@ -19,4 +19,4 @@ INTERFACE zif_abapgit_gui_event
     RAISING
       zcx_abapgit_exception.
 
-ENDINTERFACE.
+ENDINTERFACE.
```
Comment (mbtools) [inline src/ui/core/zif_abapgit_gui_event.intf.abap:22]: will the missing newline cause a diff?

File: src/ui/zif_abapgit_html_viewer.intf.abap [hunk 2]
```diff
@@ -41,4 +49,4 @@ INTERFACE zif_abapgit_html_viewer
       VALUE(rv_url) TYPE w3url.
   METHODS back .
   METHODS set_visiblity IMPORTING iv_visible TYPE abap_bool.
-ENDINTERFACE.
+ENDINTERFACE.
```
Comment (mbtools) [inline src/ui/zif_abapgit_html_viewer.intf.abap:52]: will the missing newline cause a diff?

File: src/ui/core/zif_abapgit_gui_event.intf.abap (hunk 1 above)
Comment (larshp) [reply src/ui/core/zif_abapgit_gui_event.intf.abap]: I wondered the same thing, but importing seems okay,
image

**Comment by mbtools** [comment]
go for it

**Pull request merged**